src/main/java/com/worldwarofants/game/logic/Traits.java

```diff
@@ -0,0 +1,42 @@
+package com.worldwarofants.game.logic;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.worldwarofants.game.logic.AntSpecies;
+
+public class Traits {
+
+    static Map<String, Class<? extends Traits>> traitMap = new HashMap<String, Class<? extends Traits>>();
```

Comment:
You are making this way more complicated than it needs to be. I respect the attempt to use maps but you can achieve the same thing with just.. static variables. Like this
``` private static Class casteless = Casteless.class; ```
Or whatever. 
And why are you putting classes in the map? Why do you specifically need the Class? I really do not see any point in this whole deal. 
Also, if you're planning on using the Traits class to extend it for making the other traits, call it Trait instead. Using plural is wrong here.

Reply:
We have to use this kind of mapping if we want to avoid using reflection. And we want to avoid using reflection.